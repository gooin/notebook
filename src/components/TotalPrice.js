import React from 'react';

const TotalPrice = ({income, outcome}) => {
    return (
        <React.Fragment>
            <h4>收入: {income}元 </h4>
            <h4>支出: {outcome}元</h4>
        </React.Fragment>
    );
};

export default TotalPrice;
