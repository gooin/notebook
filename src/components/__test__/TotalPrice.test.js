import React from 'react';
import {shallow} from 'enzyme';
import TotalPrice from "../TotalPrice";

const props = {
    income: 1000,
    outcome: 2000
}

describe('test TotalPrice component',()=>{
    const wrapper = shallow(<TotalPrice {...props} />)
    it('should render text correct',()=>{
        expect(wrapper.first('h4').text()).toContain(1000)
        expect(wrapper.first('h4').text()).toContain(2000)
        expect(wrapper.first('h4').text()).toContain("å…ƒ")
    })
    it('check snapshot',()=>{
        expect(wrapper.debug()).toMatchSnapshot()
    })

})