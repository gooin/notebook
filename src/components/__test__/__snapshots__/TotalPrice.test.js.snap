// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TotalPrice component check snapshot 1`] = `
"<Fragment>
  <h4>
    收入: 
    1000
    元 
  </h4>
  <h4>
    支出: 
    2000
    元
  </h4>
</Fragment>"
`;
